vortex-datafusion/src/persistent/source.rs

```diff
@@ -32,6 +32,7 @@ pub struct VortexSource {
     pub(crate) arrow_schema: Option<SchemaRef>,
     pub(crate) metrics: VortexMetrics,
     _unused_df_metrics: ExecutionPlanMetricsSet,
+    layout_readers: DashMap<Path, Arc<dyn LayoutReader>>,
```

Comment:
The lifetime of this cache is just `VortexSource`, which is initialized per query so we can just use the path as the key, without m_time